packages/gateway/src/structures/ShardingManager.ts

```diff
@@ -0,0 +1,122 @@
+import { Collection } from '@ivycord-js/utils';
+
+import { Gateway } from './Gateway';
+import { Shard } from './Shard';
+import { RawData } from 'ws';
+
+/**
+ * Represents a Discord shard manager.
+ */
+class ShardingManager {
+  /**
+   * The gateway that manages this sharding manager.
+   */
+  public gateway: Gateway;
+
+  /**
+   * A collection of all shards.
+   */
+  public shards: Collection<number, Shard>;
+
+  /**
+   * The number of shards that this sharding manager manages.
+   */
+  public shardCount = 0;
+
+  /**
+   * The shard IDs that this sharding manager manages.
+   */
+  private shardIDs: number[] = [];
+
+  /**
+   * Creates a new instance of the sharding manager.
+   * @param gateway The gateway that manages this sharding manager.
+   */
+  constructor(gateway: Gateway) {
+    this.gateway = gateway;
+    this.shards = new Collection<number, Shard>();
+  }
+
+  /**
+   * Returns the shard IDs that this sharding manager manages.
+   * @returns The shard IDs.
+   */
+  async getShardIDs() {
+    const shardIDs: number[] = [];
+    await this.gateway.fetchGatewayData();
+    if (this.gateway.shardCount === 'auto') {
+      for (let i = 0; i < this.gateway._gatewayData!.shards; i++) {
+        shardIDs.push(i);
+      }
+      this.shardCount = this.gateway._gatewayData!.shards;
+      return shardIDs;
+    }
+    for (let i = 0; i < this.gateway.shardCount; i++) {
```

Comment:
Maybe we should make a startAt option (this.gateway.shardStartAt) instead of 0?

Reply:
"shardsStartFrom" property was added for now. We'll see how that goes.